<?xml version="1.0"?>
<launch>
  <arg name="joy_config" default="ps4" />
  <arg name="joy_dev" default="/dev/input/js0" />
  <arg name="config_filepath" default="$(find aero_teleop)/config/$(arg joy_config).config.yaml" />

  <node name="joy_node"
        pkg="joy" type="joy_node" >
    <param name="dev" value="$(arg joy_dev)" />
    <param name="deadzone" value="0.3" />
    <param name="autorepeat_rate" value="20" />
  </node>

  <node name="joy_msg_switcher"
        pkg="jsk_aero_startup" type="joy_msg_switcher.py"
        output="log">
    <remap from="~angle_mode" to="/joy_msg_switcher/angle_pub" />
    <remap from="~ik_mode" to="/joy_msg_switcher/ik_pub" />
  </node>

  <node name="aero_teleop_ik"
        pkg="jsk_aero_startup"  type="aero-teleop.l"
        output="log">
    <remap from="/joy" to="/joy_msg_switcher/ik_pub" />
  </node>

  <node name="teleop_joy"
        pkg="aero_teleop"  type="xbox_one_teleop_node"
        output="log">
    <rosparam command="load" file="$(arg config_filepath)" />
    <remap from="/cmd_vel" to="/teleop/cmd_vel" />
    <remap from="/joy" to="/joy_msg_switcher/angle_pub" />
  </node>

</launch>
